watchtower-plugin/src/retrier.rs

```diff
@@ -265,6 +337,28 @@ impl Retrier {
 
         Ok(())
     }
+
+    /// For failed retriers, we set the tower to unreachable at drop time and not after backoff is complete.
+    /// Because if we set the tower as unreachable after backoff is complete, the user might perform manual retry
+    /// for that tower but this action will never be performed nor that we give error to the user.
+    /// This is because a failed retrier is never re-started, and is removed immediately by the retry manager
+    /// when it wakes from sleep (even if that failed retrier had some pending appointments).
+    ///
+    /// Setting tower status as unreachable at drop time keeps the tower in temporary unreachable state
+    /// in any other time, so it gives back an error when it is manually retried by the user.
```

Comment:
I think this is way too descriptive.

The docs should read a 1 / 2 lines describing what the method does (setting the status to unreachable) and then having a longer description is needed after a two lines break, that way the longer description is only shown if the method is expanded on the docs.

With respect to the longer description, I think you don't need to motivate it that much, it has already been mentioned twice elsewhere in the codebase why the status is not set there (which we are kind off inclined at the moment given we're moving that piece of code, but it may not be that noticeable for new contributors).

Just saying this is supposed to be done after the retrier is dropped so the issue with the user trying to retry is avoided should be enough.

Reply:
I am sometimes extra descriptive cuz my memory suck :disappointed:. Will fix that now.